usr/src/uts/common/fs/zfs/dsl_scan.c

```diff
@@ -1446,6 +1446,10 @@ dsl_scan_active(dsl_scan_t *scn)
 	return (used != 0);
 }
 
+/*
+ * Called whenever a txg syncs.  Called for both current txgs and old ones that
+ * are being resilvered
```

Comment:
I don't understand that.  This function is called every txg, "on" the current txg (meaning that tx->tx_txg == spa_syncing_txg).  It scrubs blocks in the missing range of the damaged vdevs, which (IIRC) never includes the current txg.